components/relevancy/src/lib.rs

```diff
@@ -47,9 +57,53 @@ impl RelevancyStore {
     ///
     ///  This method may execute for a long time and should only be called from a worker thread.
     #[handle_error(Error)]
-    pub fn ingest(&self, _top_urls_by_frecency: Vec<String>) -> ApiResult<()> {
+    pub fn ingest(&self, top_urls_by_frecency: Vec<String>) -> ApiResult<HashMap<Interest, u32>> {
         populate_interests::ensure_interest_data_populated(&self.db)?;
-        todo!()
+        let mut interest_counter: HashMap<Interest, u32> = HashMap::new();
+
+        let hashed_top_urls: Vec<UrlHash> = top_urls_by_frecency
+            .into_iter()
+            .filter_map(|url| hash_url(&url))
+            .collect();
+
+        let classified_urls = RelevancyStore::get_url_interest_data();
+        for classified_url in classified_urls {
+            if let Some(url) = &classified_url.url {
+                match hashed_top_urls.contains(url) {
```

Comment:
I feel like this should be reversed.  We should be looping over the top urls, and looking up them up in in the classified URLs hashmap (or database, see below).

I'm not totally sure of the logic though.  What would happen if you added `https://hello2.com` to the list of top URLs in the test below?  I think that should increase the inconclusive count by one, but the counts will stay the same.